.github/workflows/e2e.yml

```diff
@@ -43,8 +43,10 @@ jobs:
           - 5432:5432
     steps:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # Pin v4.1.1
-      # Disabled for now as it makes test runs take longer
-      # - uses: bmwill/rust-cache@v1 # Fork of 'Swatinem/rust-cache' which allows caching additional paths
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-all-crates: true
+          cache-on-failure: true
```

Comment:
this shaves a few minutes off the build, but doesn't seem to cache builds for things in the repo.  Probably needs further improvement